src/main/java/org/kohsuke/github/GHRepositoryStatistics.java

```diff
@@ -0,0 +1,444 @@
+
+        @Override
+        public URL getHtmlUrl() throws IOException {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        public GitHub getRoot() {
+            return root;
+        }
+
+        /**
+         * @return The list of commit counts for everyone combined, for the
+         * last 52 weeks.
+         */
+        public List<Integer> getAllCommits() {
+            return all;
+        }
+
+        /**
+         * @return The list of commit counts for the owner, for the
+         * last 52 weeks.
+         */
+        public List<Integer> getOwnerCommits() {
+            return owner;
+        }
+
+        /*package*/ Participation wrapUp(GitHub root) {
+        this.root = root;
+        return this;
+        }
+    }
+
+    /**
+     * Get the number of commits per hour in each day.
+     * See https://developer.github.com/v3/repos/statistics/#get-the-number-of-commits-per-hour-in-each-day
+     */
+    public List<PunchCardItem> getPunchCard() throws IOException {
+        // Map to ArrayLists first, since there are no field names in the
+        // returned JSON.
+        InputStream stream = root.retrieve().asStream(getApiTailUrl("stats/punch_card"));
+
+        ObjectMapper mapper = new ObjectMapper();
+        TypeReference<ArrayList<ArrayList<Integer> > > typeRef =
+                new TypeReference<ArrayList< ArrayList<Integer> > >() {};
+        ArrayList<ArrayList <Integer> > list = mapper.readValue(stream, typeRef);
+
+        // Convert to proper objects.
+        ArrayList<PunchCardItem> returnList = new ArrayList<PunchCardItem>();
+        for(ArrayList<Integer> item: list) {
+            PunchCardItem pci = new PunchCardItem(item);
+            returnList.add(pci);
+        }
+
+        return returnList;
+    }
+
+    public static class PunchCardItem {
+        private int dayOfWeek;
+        private int hourOfDay;
+        private int numberOfCommits;
+
+        private PunchCardItem(ArrayList<Integer> item) {
+            dayOfWeek = item.get(0);
+            hourOfDay = item.get(1);
+            numberOfCommits = item.get(2);
+        }
+
+        /**
+         * @return The day of the week.
+         * 0 = Sunday, 1 = Monday, etc.
+         */
+        public int getDayOfWeek() {
+            return dayOfWeek;
+        }
+
+        /**
+         * @return The hour of the day from 0 to 23.
+         */
+        public long getHourOfDay() {
+            return hourOfDay;
+        }
+
+        /**
+         * @return The number of commits for the day and hour.
+         */
+        public long getNumberOfCommits() {
+            return numberOfCommits;
+        }
+
+        public String toString() {
+            return "Day " + getDayOfWeek() + " Hour " + getHourOfDay() + ": " +
+                    getNumberOfCommits() + " commits";
+        }
+    }
+
+    String getApiTailUrl(String tail) {
+        return repo.getApiTailUrl(tail);
```

Comment:
All of the children are under `stats/`, so that string should move in here. 
```suggestion
        return repo.getApiTailUrl("stats/" + tail);
```

Reply:
Thanks, I've made this change.